packages/app/src/cli/services/dev/output.ts

```diff
@@ -16,45 +16,61 @@ export async function outputUpdateURLsResult(
   if (app.newApp) {
     renderInfo({
       headline: `For your convenience, we've given your app a default URL: ${urls.applicationUrl}.`,
-      body: `You can update your app's URL anytime in the ${dashboardURL}. But once your app is live, updating its URL will disrupt merchant access.`,
+      body: [
+        "You can update your app's URL anytime in the",
+        dashboardURL,
+        'But once your app is live, updating its URL will disrupt merchant access.',
+      ],
     })
   } else if (!updated) {
     renderInfo({
-      headline: `To make URL updates manually, you can add the following URLs as redirects in your ${dashboardURL}:`,
+      headline: [
+        'To make URL updates manually, you can add the following URLs as redirects in your',
+        dashboardURL,
+        {char: ':'},
+      ],
       body: {list: {items: urls.redirectUrlWhitelist}},
     })
   }
 }
 
-export function outputAppURL(storeFqdn: string, url: string) {
-  const title = url.includes('localhost') ? 'App URL' : 'Shareable app URL'
-  const heading = output.token.heading(title)
-  const appURL = buildAppURLForWeb(storeFqdn, url)
-  output.info(output.content`\n\n${heading}\n\n  ${appURL}\n`)
+interface OutputPreviewUrlOptions {
+  app: AppInterface
+  storeFqdn: string
+  exposedUrl: string
+  proxyUrl: string
+  appPreviewAvailable: boolean
 }
 
-export function outputDevConsoleURL(url: string) {
-  const title = 'Shopify extension dev console URL'
-  const heading = output.token.heading(title)
-  const devConsoleURL = `${url}/extensions/dev-console`
-  output.info(output.content`${heading}\n\n  ${devConsoleURL}\n`)
-}
+export function outputPreviewUrl({app, storeFqdn, exposedUrl, proxyUrl, appPreviewAvailable}: OutputPreviewUrlOptions) {
+  if (!appPreviewAvailable && app.extensions.ui.length === 0) {
```

Comment:
in which cases will `appPreviewAvailable` be false? shouldn't we always be able to `dev` the app?

Reply:
@isaacroldan do you think it still makes sense to show the dev console url if they have decided not to update the partners URLs? I think they wouldn't be able to see the dev console in that case, but the previous logic didn't account for that. Maybe we haven't gotten any bug reports because few people were going to the dev console with just `localhost`.

Reply:
I've greatly simplified the logic, could you please take another look?

Reply:
The dev console doesn't depend on the URLs being updated (the extensions should work anyway), so if there are extensions i'd say we need to show the url always